packages/lesswrong/components/votes/lwReactions/NamesAttachedReactionsVoteOnComment.tsx

```diff
@@ -297,18 +266,22 @@ export const useNamesAttachedReactionsVoting = (voteProps: VotingProps<VoteableT
     return alreadyUsedReactionTypesByKarma
   }
 
-  function getAllReactionQuotes() {
-    const alreadyUsedReactions = getAlreadyUsedReacts()
-    const allReacts = Object.values(alreadyUsedReactions)
-    const allQuotes = allReacts.flatMap(r => r?.flatMap(r => r?.quotes ?? [])).filter(q => q !== undefined) as string[]
-    return uniq(allQuotes)
-  }
-
   return {
-    currentUserExtendedVote, getCurrentUserReaction, getCurrentUserReactionVote, toggleReaction, setCurrentUserReaction, getAlreadyUsedReactTypesByKarma, getAlreadyUsedReacts, getAllReactionQuotes
+    getCurrentUserReaction, getCurrentUserReactionVote, toggleReaction, setCurrentUserReaction, getAlreadyUsedReactTypesByKarma, getAlreadyUsedReacts
   };
 }
 
+export function reactionVoteIsMatch(react: UserVoteOnSingleReaction, name: EmojiReactName, quote: QuoteLocator|null): boolean {
+  if (react.react !== name)
+    return false;
+  if (quote) {
+    return !!(react.quotes && react.quotes!.indexOf(quote)>=0);
```

Comment:
re: `react.quotes!.indexOf` - any reason that isn't `react.quotes.indexOf` instead, having already put it after `react.quotes &&`?

Reply:
I'm not sure how that `!` got there; it doesn't seem to be needed now (but presumably was needed at some point and then not removed when it became spurious). Will remove.